### PR TITLE
layers: Fix std::optional crash

### LINKED ISSUE #4853: assertion this->has_value() failed: optional operator* called on a disengaged value crash
**Describe the Issue**

The synchronization_validation.cpp code can trigger a std::optional crash
```
./../buildtools/third_party/libc++/trunk/include/optional:953: assertion this->has_value() failed: optional operator* called on a disengaged value
```

likely introduced by https://github.com/KhronosGroup/Vulkan-ValidationLayers/commit/a8564aa133bf79a09ff6a240de2a696930e07f33

as shown by the backtrace
```
../../buildtools/third_party/libc++/trunk/include/optional:953: assertion this->has_value() failed: optional operator* called on a disengaged value
Thread 1 "dawn_end2end_te" received signal SIGABRT, Aborted.
__pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at ./nptl/pthread_kill.c:44
44	./nptl/pthread_kill.c: No such file or directory.
(gdb) bt
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0)
    at ./nptl/pthread_kill.c:44
#1  0x00007ffff7a895df in __pthread_kill_internal (signo=6, threadid=<optimized out>)
    at ./nptl/pthread_kill.c:78
#2  0x00007ffff7a3da02 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff7a28469 in __GI_abort () at ./stdlib/abort.c:79
#4  0x00007fffe6999322 in std::Cr::__libcpp_verbose_abort(char const*, ...) ()
    at ../../buildtools/third_party/libc++/trunk/src/verbose_abort.cpp:74
#5  0x00007fffe65a015c in ValidateLoadOperation ()
    at ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/synchronization_validation.cpp:1408
#6  0x00007fffe65b503d in Validate ()
    at ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/synchronization_validation.cpp:7431
#7  0x00007fffe65b4e36 in ValidateBeginRenderPass ()
    at ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/synchronization_validation.cpp:4712
#8  0x00007fffe65b51a0 in PreCallValidateCmdBeginRenderPass ()
    at ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/synchronization_validation.cpp:4722
#9  0x00007fffe68a8c43 in CmdBeginRenderPass ()
    at ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/generated/chassis.cpp:3685
// non vulkan validation layers stuff below
#10 0x0000555555d0846d in RecordBeginRenderPass ()
    at ../../third_party/dawn/src/dawn/native/vulkan/CommandBufferVk.cpp:341
#11 RecordRenderPass () at ../../third_party/dawn/src/dawn/native/vulkan/CommandBufferVk.cpp:1058
#12 0x0000555555d07758 in RecordCommands ()
    at ../../third_party/dawn/src/dawn/native/vulkan/CommandBufferVk.cpp:739
#13 0x0000555555d15db1 in SubmitImpl () at ../../third_party/dawn/src/dawn/native/vulkan/QueueVk.cpp:54
#14 0x0000555555ca59ff in SubmitInternal () at ../../third_party/dawn/src/dawn/native/Queue.cpp:538
#15 0x0000555555ca5875 in dawn::native::QueueBase::APISubmit(unsigned int, dawn::native::CommandBufferBase* const*) () at ../../third_party/dawn/src/dawn/native/Queue.cpp:197
#16 0x00005555559bc286 in InitializeDepthStencilTextureRegion ()
    at ../../third_party/dawn/src/dawn/tests/end2end/DepthStencilCopyTests.cpp:180
#17 0x00005555559b6286 in DoCopyFromStencilTest ()
```

and further trace down to 
```
../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/synchronization_validation.cpp:7940
7940	            got = (view_mask_ == VK_IMAGE_ASPECT_DEPTH_BIT) ? &(*gen_store_[Gen::kRenderArea])
at ../../third_party/vulkan-deps/vulkan-validation-layers/src/layers/synchronization_validation.cpp:7941
7941	                                                            : &(*gen_store_[Gen::kDepthOnlyRenderArea]);
```

This issue is found in vulkan-deps roll in Chromium and triggered by a dawn test. More details for reproduce: https://crbug.com/1385811#c7

**Valid Usage ID**
If applicable, please include the validation messages encountered leading up to the issue
(Nop it was a assert failure crash)

**Environment:**
 - OS:  Linux
 - GPU: NVIDIA_Quadro_P1000
 - SDK or header version if building from repo: bd2c589d0d34612d9dd071396225d7de772c020c
 - Options enabled (synchronization, best practices, etc.):

**Additional context**

### PR DESCRIPTION
**Pull request by juan-lunarg**
Currently code doesn't check if std::optional has a value

closes #4853

### CODE REVIEW

File: layers/synchronization_validation.cpp [hunk 1]
```diff
@@ -7927,27 +7927,14 @@ AttachmentViewGen::AttachmentViewGen(const IMAGE_VIEW_STATE *view, const VkOffse
...
-        case kViewSubresource:
-            got = &(*gen_store_[kViewSubresource]);
-            break;
-        case kRenderArea:
-            got = &(*gen_store_[kRenderArea]);
-            break;
-        case kDepthOnlyRenderArea:
-            got = (view_mask_ == VK_IMAGE_ASPECT_DEPTH_BIT) ? &(*gen_store_[Gen::kRenderArea])
-                                                            : &(*gen_store_[Gen::kDepthOnlyRenderArea]);
-            break;
```
Comment (juan-lunarg) [inline layers/synchronization_validation.cpp:7942]: @jzulauf-lunarg I don't particularly understand the logic here. What is the explanation?

**Comment by spencer-lunarg** [comment]
LGTM! Pending Response about the `GetRangeGen` function

**Review by arno-lunarg — COMMENTED** [review]
LGTM! Same question as you and Spencer

File: layers/synchronization_validation.cpp (hunk 1 above)
Comment (jzulauf-lunarg) [reply layers/synchronization_validation.cpp:7942]: @juan-lunarg  If the view is a depth only view, then the depth only portion of the render area is simply the render area.  If the view is a depth stencil view, then the depth only portion of the render area will be a subset, and thus needs the generator function that will produce the address ranges of that subset

**Review by arno-lunarg — APPROVED** [review]

**Pull request merged**